cmd/build.go

```diff
@@ -110,47 +112,73 @@ func Build(ctx context.Context) *cobra.Command {
 	return cmd
 }
 
-func buildV2(m model.ManifestBuild, options build.BuildOptions, args []string) error {
+func keepOnlySelectedServices(service string, manifest *model.ManifestBuild) {
+	for key := range *manifest {
+		if key != service {
+			delete(*manifest, key)
+		}
+	}
+}
+
+func buildV2(manifest model.ManifestBuild, options build.BuildOptions, args []string) error {
 	service := ""
 	if len(args) == 1 {
 		service = args[0]
 	}
 
+	// settings for single build
 	if service != "" {
-		buildInfo, ok := m[service]
+		_, ok := manifest[service]
 		if !ok {
 			return fmt.Errorf("invalid service name: %s", service)
 		}
-		if !okteto.Context().IsOkteto && buildInfo.Image == "" {
-			return fmt.Errorf("image tag is required when context is not okteto")
-		}
+
+		keepOnlySelectedServices(service, &manifest)
```

Comment:
Maps are references, like pointers or slices. Getting a pointer of it is not needed when working with functions. This also means `*` at `keepOnlySelectedServices` can be removed.